src/plugin.rs

```diff
@@ -0,0 +1,214 @@
+use std::str::FromStr;
+use std::sync::Arc;
+
+use pyo3::{prelude::*, types::PyType};
+
+use crate::{DecryptError, EncryptError, IdentityError, RecipientError};
+
+#[pyclass(module = "pyrage.plugin", subclass)]
+#[derive(Clone)]
+pub(crate) struct Callbacks;
+
+/// Hack, because the orphan rule would prevent us from deriving a
+/// foreign trait on a foreign object. Instead, define a newtype.
+#[derive(Clone)]
+pub(crate) struct PyCallbacks(Py<Callbacks>);
+
+#[pymethods]
+#[allow(unused_variables)]
+impl Callbacks {
```

Comment:
Just leaving a note: I need to think about this a bit. Rather than messing around with subclassing like this, we may just want the newtype to be over `Py<PyAny>` and do structural subtyping instead (since we get relatively little from the concrete base here).

Reply:
> I suppose I could rewrite this bit to duck-type on Rust side, and add an interface class to the Python type stubs so that the Python side can still type-check the usage of this API.

Yeah, this sounds good to me -- I think the performance will be identical, and will be a bit easier to reason about/read here :slightly_smiling_face: